File: client/android/app/src/main/java/com/shadangakriya/app/ScreenProtectionPlugin.java
package com.shadangakriya.app;

import android.view.WindowManager;
import android.widget.Toast;
import com.getcapacitor.Plugin;
import com.getcapacitor.PluginCall;
import com.getcapacitor.PluginMethod;
import com.getcapacitor.annotation.CapacitorPlugin;

@CapacitorPlugin(name = "ScreenProtection")
public class ScreenProtectionPlugin extends Plugin {

    @PluginMethod
    public void enableProtection(PluginCall call) {
        getActivity().runOnUiThread(() -> {
            getActivity().getWindow().setFlags(
                WindowManager.LayoutParams.FLAG_SECURE,
                WindowManager.LayoutParams.FLAG_SECURE
            );
            call.resolve();
        });
    }

    @PluginMethod
    public void disableProtection(PluginCall call) {
        getActivity().runOnUiThread(() -> {
            getActivity().getWindow().clearFlags(WindowManager.LayoutParams.FLAG_SECURE);
            call.resolve();
        });
    }

    @PluginMethod
    public void showToast(PluginCall call) {
        String message = call.getString("message", "Screen recording is not allowed");
        getActivity().runOnUiThread(() -> {
            Toast.makeText(getContext(), message, Toast.LENGTH_LONG).show();
            call.resolve();
        });
    }
}

File: client/ios/App/App/ScreenProtectionPlugin.m
#import <Foundation/Foundation.h>
#import <Capacitor/Capacitor.h>

CAP_PLUGIN(ScreenProtectionPlugin, "ScreenProtection",
    CAP_PLUGIN_METHOD(enableProtection, CAPPluginReturnPromise);
    CAP_PLUGIN_METHOD(disableProtection, CAPPluginReturnPromise);
    CAP_PLUGIN_METHOD(showToast, CAPPluginReturnPromise);
    CAP_PLUGIN_METHOD(isScreenBeingCaptured, CAPPluginReturnPromise);
)
